### DISCUSSION TITLE
Best way to implement hook for add ons

### DISCUSSION THREAD

**Discussion by sethstha** [WordPress/gutenberg#38559] [Q&A] [answered]
As on the PHP we used to leave some `do_action` to let third-party plugins render some HTML. 

As `doAction` doesn't return while `applyFilters` does on javascript. Now on the react side, it needs to return to the display component. While I use applyFilters older one gets completely replaced.

Is there a better way to leave hooks to let others render their HTML on react side?

**Comment by Mamaduka** [comment, marked as answer]
Hi, @sethstha

I would recommend looking into [SlotFills](https://developer.wordpress.org/block-editor/reference-guides/slotfills/) API. It is what core uses to allow plugins to render their components in the UI.

**Reply by sethstha** [reply]
And what about when we separate the javascript files Like on Addon for the plugin?

**Reply by Mamaduka** [reply]
Can you provide more details? I'm not sure I understand the question.

**Reply by sethstha** [reply]
So we built a plugin over React that uses most of the packages from WordPress, It is also on typescript so the types are coming from React. Now we want to build an addon for the plugin that will extend the UI on the settings of our plugin. So the problem here is we will have two different js files and to be able to use Fill, Slot we need to wrap that on same `SlotFillProvider`. I tried extending with `withFilter`. But no luck.
